gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/launcher/ProcessWorkUnitsJobLauncher.java

```diff
@@ -93,4 +97,11 @@ public void submitJob(List<WorkUnit> workunits) {
       throw new RuntimeException(e);
     }
   }
+
+  private void setupMDCFromJobState(JobState jobState) {
```

Comment:
I expect this will be reused for other job launchers, even if we're merely creating them during dev (like the GenWUs I'm about to release).  to support that, maybe situate in `Help` or `JobStateUtils`?

and since I believe this MDC stuff all happens transparently behind the scenes, maybe less impl jargon in the name.  e.g. `propagateGaaSFlowExecutionContext` or `initFlowExecutionContext`?